### PR TITLE
Throw the IllegalArgumentException if the address is empty to avoid NPE.

### PR DESCRIPTION
**Pull request by MartianQiu**
Throw the IllegalArgumentException if the address is empty to avoid NPE.

### CODE REVIEW

**Review by horizonzy — APPROVED** [review]
LGTM.

**Review by xiaoheng1 — APPROVED** [review]
LGTM

**Comment by horizonzy** [comment]
Some unit test can't pass, should modify it to suit new logic.

**Review by horizonzy — COMMENTED** [review]
Some unit test can't pass, should modify it to suit new logic.

**Comment by AlbumenJ** [comment]
Please fix unit test.

**Comment by MartianQiu** [comment]
I have fixed these unit tests, please check it again, thanks a lot!

File: dubbo-common/src/test/java/org/apache/dubbo/common/utils/UrlUtilsTest.java [hunk 1]
```diff
@@ -334,8 +344,8 @@ public void testIsItemMatch() throws Exception {
     public void testIsServiceKeyMatch() throws Exception {
         URL url = URL.valueOf("test://127.0.0.1");
         URL pattern = url.addParameter(GROUP_KEY, "test")
-                .addParameter(INTERFACE_KEY, "test")
```
Comment (horizonzy) [inline dubbo-common/src/test/java/org/apache/dubbo/common/utils/UrlUtilsTest.java:337]: line indent should be fix.

**Review by horizonzy — COMMENTED** [review]
Code for

**Pull request merged**